usaspending_api/search/v2/views/spending_by_category.py

```diff
@@ -0,0 +1,205 @@
+import logging
+
+from django.db.models import F, Sum
+
+# from usaspending_api.awards.v2.filters.sub_award import subaward_filter
+from usaspending_api.awards.v2.filters.view_selector import spending_by_category as sbc_view_queryset
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.common.helpers import get_simple_pagination_metadata
+from usaspending_api.core.validator.award_filter import AWARD_FILTER
+from usaspending_api.references.models import Cfda
+
+logger = logging.getLogger(__name__)
+
+
+class business_logic:
+    __slots__ = ('category', 'scope', 'page', 'limit', 'lower_limit', 'upper_limit', 'filters', 'queryset', 'model', )
+
+    def __new__(cls, payload):
+        """
+        Use __new__ instead of __init__ since __new__ can return a value other than None.
+        Since this is a class method using common nomenclature for class: `cls`
+        """
+        cls.category = payload['category']
+        cls.scope = payload['scope']
+        cls.page = payload['page']
+        cls.limit = payload['limit']
+
+        cls.lower_limit = (cls.page - 1) * cls.limit
+        cls.upper_limit = cls.page * cls.limit
+
+        cls.filters = {
+            item['name']: payload[item['name']] for item in AWARD_FILTER if item['name'] in payload}
+
+        if (cls.scope is None) and (cls.category != "cfda_programs"):
```

Comment:
CFDA programs do not need a scope right? I believe TinyShield has scope as required because when I try `category:cfda_programs` the endpoint returns a 422 error
```
{
    "detail": "Missing value: 'scope' is a required field"
}
```

Reply:
I forgot "required" is the default value (`optional` = False). Easy to change